File: backend/app/models/item.py
# today_book/backend/app/models/item.py
from sqlalchemy import Column, Integer, String, Text, Float, Boolean, Date, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from datetime import datetime, timezone
from app.database import Base


class ItemCategory(Base):
    __tablename__ = "item_categories"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    name = Column(String(100), nullable=False)
    parent_id = Column(Integer, ForeignKey("item_categories.id"))
    sort_order = Column(Integer, default=0)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    items = relationship("Item", back_populates="category")


class ItemLocation(Base):
    __tablename__ = "item_locations"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    name = Column(String(100), nullable=False)
    parent_id = Column(Integer, ForeignKey("item_locations.id"))
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    items = relationship("Item", back_populates="location")


class Item(Base):
    __tablename__ = "items"

    id = Column(Integer, primary_key=True, autoincrement=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    name = Column(String(200), nullable=False)
    description = Column(Text)
    category_id = Column(Integer, ForeignKey("item_categories.id"))
    location_id = Column(Integer, ForeignKey("item_locations.id"))
    purchase_date = Column(Date)
    purchase_price = Column(Float)
    current_value = Column(Float)
    quantity = Column(Integer, default=1)
    is_consumable = Column(Boolean, default=False)
    warranty_expire = Column(Date)
    notes = Column(Text)
    asset_tag = Column(String(50))
    is_archived = Column(Boolean, default=False)
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))
    updated_at = Column(DateTime, default=lambda: datetime.now(timezone.utc), onupdate=lambda: datetime.now(timezone.utc))

    user = relationship("User", back_populates="items")
    category = relationship("ItemCategory", back_populates="items")
    location = relationship("ItemLocation", back_populates="items")
    attachments = relationship("ItemAttachment", back_populates="item", cascade="all, delete-orphan")


class ItemAttachment(Base):
    __tablename__ = "item_attachments"

    id = Column(Integer, primary_key=True, autoincrement=True)
    item_id = Column(Integer, ForeignKey("items.id"), nullable=False)
    filename = Column(String(255), nullable=False)
    filepath = Column(String(500), nullable=False)
    attachment_type = Column(String(20))  # photo/receipt/warranty
    created_at = Column(DateTime, default=lambda: datetime.now(timezone.utc))

    item = relationship("Item", back_populates="attachments")

File: frontend/src/api/item.ts
/* today_book/frontend/src/api/item.ts */
import apiClient from './client'

export interface ItemCategory {
  id: number
  name: string
  parent_id?: number
}

export interface ItemLocation {
  id: number
  name: string
  parent_id?: number
}

export interface Item {
  id: number
  name: string
  description?: string
  category_id?: number
  location_id?: number
  purchase_date?: string
  purchase_price?: number
  current_value?: number
  quantity: number
  is_consumable: boolean
  warranty_expire?: string
  notes?: string
  asset_tag?: string
  is_archived: boolean
  created_at: string
}

export const itemApi = {
  list: (params?: Record<string, unknown>) => apiClient.get<Item[]>('/items/', { params }),
  get: (id: number) => apiClient.get<Item>(`/items/${id}`),
  create: (data: Partial<Item>) => apiClient.post<Item>('/items/', data),
  update: (id: number, data: Partial<Item>) => apiClient.put<Item>(`/items/${id}`, data),
  delete: (id: number) => apiClient.delete(`/items/${id}`),
  getCategories: () => apiClient.get<ItemCategory[]>('/items/categories'),
  createCategory: (data: { name: string; parent_id?: number }) => apiClient.post('/items/categories', data),
  getLocations: () => apiClient.get<ItemLocation[]>('/items/locations'),
  createLocation: (data: { name: string; parent_id?: number }) => apiClient.post('/items/locations', data),
  getStats: () => apiClient.get('/items/stats/summary'),
}
